### DISCUSSION TITLE
Zeal / Dash Docset for Janet

### DISCUSSION THREAD

**Discussion by CosmicToast** [janet-lang/janet#1357] [Show and tell]
[Zeal](https://zealdocs.org) / [Dash](https://kapeli.com/dash) are an offline documentation indexing and browsing system with full text fuzzy searching capabilities.

I recently wrote a generator for a Janet docset, the config for which you can find [here](https://tilde.toast.cafe/~toast/dash/Janet.docsetconfig) (all my links likely not permanent).
A prebuilt version is [available](https://tilde.toast.cafe/~toast/dash/Janet.docset.zip) as well.

I found myself having janet docs open in many different tabs with no good searching or jumping, so this should help that.
It seems like there are even more programs that can use the docset format as well.

**Comment by iacore** [comment]
`many different tabs`

FYI, there's the [API page](https://janet-lang.org/api/index.html), and `(doc doc)`

**Reply by CosmicToast** [reply]
Indeed, the API page is what I'm generating the docs from, it's what I mentioned re: "Janet docs open in many tabs".
And the reason to have a docset is indeed to allow for faster searching and jumping.

An LSP can get a good chunk of the way there, and it's less hard to make one with `doc doc`, but none of these really solves the discoverability problem that FTS and fuzzy searching can help with.

**Comment by sogaiu** [comment]
I think as pointed out in the original posting, the UI (searching and jumping) for Zeal provides something over the web browser / repl experience.

I did find getting things to work in zeal to be on the confusing side.  In the end I copied the uncompressed docset under `~/.local/share/Zeal/Zeal/docsets/` (don't know why it's there and not `~/.local/share/zeal/docsets`, but perhaps that's an OS-dependent thing?).  Somehow the `Add feed` in the `Docsets` dialog that appears via `Tools -> Docsets` wasn't leading to success.

May be an icon can be added?  That's about the only thing [my older less complete effort](https://gitlab.com/sogaiu/janet-zeal-docset) seems to bring to the table (^^;

**Reply by CosmicToast** [reply]
There already is an icon, at least on my machine: 
<img width="88" alt="image" src="https://github.com/janet-lang/janet/assets/5273047/46302c78-e63c-4354-bb08-f2c58f196294">

It doesn't seem to show up in the docset listing though, I'm not sure why.
It might be because it can't handle .tiff?

**Reply by sogaiu** [reply]
Weird, it didn't show up for me:

![zeal-no-janet-icon](https://github.com/janet-lang/janet/assets/33044872/a894e87e-1037-4b58-84ee-9699e4ea7399)

I tried copying [this `.png`](https://gitlab.com/sogaiu/janet-zeal-docset/-/blob/2827c4cda1195fd29b1ae8c9b68ae3f72b0eec90/janet-no-bg-32x32.png) into the same directory that had `icon.tiff` and renaming it to `icon.png`.

That worked for me.

---

In any case, nice job on the docset!

**Reply by CosmicToast** [reply]
That sounds like a Zeal bug then, since tiff should be fine afaik.
The docsetconfig format (afaik, it's not documented yet, ironically enough) doesn't seem to allow for custom handling there.

**Reply by sogaiu** [reply]
Regarding the icon, perhaps as you say, it is a zeal bug.

May be you took this in already, but it seems that in the "Contribute a new docset" section for Dash that you posted a [link](https://github.com/Kapeli/Dash-User-Contributions#contribute-a-new-docset) to, there is this:

```
Include an icon.png and icon@2x.png with sizes 16x16 and 32x32 or simply delete the sample icon if you don't want an icon at all
```

So may be it's worth having some `.png` anyway?

**Reply by CosmicToast** [reply]
While I did mention before that I'd like to limit post-build steps I have to take, for the time being I just edited the .zip.
Feel free to re-download it and it should have both of those (simply extracted from the .tiff, pages 4 and 3 respectively). :)

**Reply by sogaiu** [reply]
Thanks for the changes.

I see a nice icon now :)

**Comment by pepe** [comment]
I am a first-time Zeal user. I like it a lot! Thank you @CosmicToast!

These things are very important for our efforts to spread Janet. It will come in handy to me when I teach Janet at the end of February :-D.

**Comment by iacore** [comment]
Is it possible to submit the docset to Dash/Zeal's list of docsets?

**Reply by CosmicToast** [reply]
Yes, the requirements are [here](https://github.com/Kapeli/Dash-User-Contributions#contribute-a-new-docset).
My main complaint is that they're oriented around someone maintaining a set of builds, which I'm not sure as to my willingness to do.
It doesn't seem like there's a docsetconfig user submission repo.
I might try to poke the Dash developer to create one, since the generic built-in docset generator is actually new from dash 7.
